### PR TITLE
Cache image

### PR DESCRIPTION
**Pull request by terovirtanen**
## Description
Create and use cache image on docker image build. Cache image contains dependency cache and .next cache.

### Closes

**[DEVOPS-498](https://helsinkisolutionoffice.atlassian.net/browse/DEVOPS-498)**

## Additional notes

[DEVOPS-498]: https://helsinkisolutionoffice.atlassian.net/browse/DEVOPS-498?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**$(upper(replace(parameters.proxy,'-', ' '))) is deployed to platta: https://venue-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**$(upper(replace(parameters.proxy,'-', ' '))) is deployed to platta: https://events-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr200.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://harrastukset-pr200.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**$(upper(replace(parameters.proxy,'-', ' '))) is deployed to platta: https://events-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**$(upper(replace(parameters.proxy,'-', ' '))) is deployed to platta: https://venue-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr200.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr200.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr200.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr200.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

File: docker-compose.hobbies.yml [hunk 1]
```diff
@@ -54,6 +54,22 @@ services:
         - APP_PORT=3000
         - PROJECT=hobbies-helsinki
 
+  # Convenience production "cache" stage service for debugging
```
Comment (nenonja) [inline docker-compose.hobbies.yml:57]: What about the other docker-compose files?

**Review by nenonja — COMMENTED** [review]
I didnt dig deeper into cache details. Just wondering do we need the same caching also for the `proxy.Dockerfile`?

**Review by nenonja — APPROVED** [review]

**Review by kristianjokela — APPROVED** [review]

File: docker-compose.hobbies.yml (hunk 1 above)
Comment (terovirtanen) [reply docker-compose.hobbies.yml:57]: Cache image support added  to sports and events docker-compose too

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://tapahtumat-pr200.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr200.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr200.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://tapahtumat-pr200.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr200.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**